src/pages/monitor/Monitor.tsx

```diff
@@ -0,0 +1,399 @@
+      ? `${data.reliability}%`
+      : (): ReactElement => (
+          <NegativeComponent disable={!isPositive() ? true : false} />
+        ),
+    stateHeight: isPositive()
+      ? `#${data.stateheight}`
+      : (): ReactElement => (
+          <NegativeComponent
+            useHashTag={true}
+            disable={!isPositive() ? true : false}
+          />
+        ),
+    isItUp: (): ReactElement => <IsItUp statusIsItUp={data.status} />,
+    chain: data.status || '',
+  }
+}
+
+const returnNodesListData = (
+  data: Array<WSDoraData>,
+  returnStub: boolean,
+  network: string,
+): Array<ParsedNodes> => {
+  if (returnStub) {
+    return MOCK_NODES.map(n => n)
+  } else {
+    return data.map(data => mapNodesData(data))
+  }
+}
+
+const columns: ColumnType[] = [
+  { name: 'Endpoint', accessor: 'endpoint', sortOpt: 'endpoint' },
+  { name: 'Type', accessor: 'type', sortOpt: 'type' },
+  { name: 'Is it up?', accessor: 'isItUp', sortOpt: 'isItUp' },
+  { name: 'Reliability', accessor: 'reliability', sortOpt: 'reliability' },
+  { name: 'State Height', accessor: 'stateHeight', sortOpt: 'stateHeight' },
+  { name: 'Block Height', accessor: 'blockHeight', sortOpt: 'blockHeight' },
+  { name: 'Version', accessor: 'version', sortOpt: 'version' },
+  { name: 'Peers', accessor: 'peers', sortOpt: 'peers' },
+]
+
+const NetworkStatus: React.FC<{}> = () => {
+  const nodes = useSelector(({ node }: { node: NodeState }) => node)
+  const BEST_BLOCK = 'Best Block'
+  const LAST_BLOCK = 'Last Block'
+  const AVG_BLOCK_TIME = 'Avg Block Time'
+
+  const [bestBlock, setBestBlock] = useState<number>(0)
+  const [lastBlockCounter, setLastBlockCounter] = useState<number>(0)
+  const [listAvgBlockTime, setListAvgBlockTime] = useState<number[]>([])
+  const [avgBlockTime, setAvgBlockTime] = useState<string>('')
+
+  const getBestBlock = (): number => {
+    const sortedList = SerializeNode(nodes).sort((data1, data2) => {
+      if (data1.height <= data2.height) {
+        return 0
+      } else return 1
+    })
+
+    if (sortedList.length > 0) {
+      return sortedList[0].height
+    } else return 0
+  }
+
+  const handleLastBlock = (): void => {
+    setInterval(() => {
+      setLastBlockCounter(prevState => {
+        const newCounter = prevState + 1
+        return newCounter
+      })
+    }, 1000)
+  }
+
+  const handleAvgBlockCounter = (): void => {
+    setListAvgBlockTime(prevState => [...prevState, lastBlockCounter])
+  }
+
+  useEffect(() => {
+    handleAvgBlockCounter()
+    setLastBlockCounter(0) //eslint-disable-next-line
+  }, [bestBlock])
+
+  useEffect(() => {
+    const sumBlockTime = listAvgBlockTime.reduce((avg, time) => {
+      avg = avg + time
+      return avg
+    }, 0)
+    setAvgBlockTime((sumBlockTime / listAvgBlockTime.length).toFixed(1))
+  }, [listAvgBlockTime])
+
+  useEffect(() => {
+    handleLastBlock()
+  }, [])
+
+  useEffect(() => {
+    setBestBlock(getBestBlock()) //eslint-disable-next-line
+  }, [nodes])
```

Comment:
Shouldnt these all be inside a single `useEffect` hook?

Reply:
I don't think so, because each hook controls a part of the function's workflow, if you group everything on a single hook, it will alter the current flow and it will cause some bugs (I already tested it)